### PR TITLE
Separate Post for Creating Issues

### PR DESCRIPTION
**Pull request by Remakh**
#### Summary
If you are subscribed to only when issues are created, you do not receive notifications for other cases.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-plugin-github/issues/390

### CODE REVIEW

File: server/plugin/command.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ import (
 )
 
 const (
+	featureIssueCreation = "issue_creation"
```
Comment (hanzei) [inline server/plugin/command.go:17]: Let's use the plural here for consistency 
```suggestion
	featureIssueCreation = "issue_creations"
```

**Review by hanzei — CHANGES_REQUESTED** [review]
Thanks for the PR :+1: 

Could you please update the autocomplete data in `command.go`?

File: server/plugin/command.go (hunk 1 above)
Comment (Remakh) [reply server/plugin/command.go]: Hello, the change has been made.

File: server/plugin/webhook.go [hunk 2]
```diff
@@ -390,6 +390,10 @@ func (p *Plugin) postIssueEvent(event *github.IssuesEvent) {
 			continue
 		}
 
+		if sub.IssueCreation() && action != "opened" {
+			continue
+		}
```
Comment (mickmister) [inline server/plugin/webhook.go:395]: Should we disallow setting both `issues` and `issue_creations` when setting up a subscription?

Comment (hanzei) [reply server/plugin/webhook.go:395]: Good point :+1:

**Review by hanzei — CHANGES_REQUESTED** [review]
Did you got the chance to take look at my comment above? https://github.com/mattermost/mattermost-plugin-github/pull/392#pullrequestreview-511694417

**Comment by Remakh** [comment]
> Did you got the chance to take look at my comment above? [#392 (review)](https://github.com/mattermost/mattermost-plugin-github/pull/392#pullrequestreview-511694417)

Yes I did. Changed it to issue_creations in my latest commit.

File: server/plugin/webhook.go (hunk 2 above)
Comment (Remakh) [reply server/plugin/webhook.go:395]: Thanks for the review! Changing it now.

**Comment by hanzei** [comment]
@Remakh I was referring to the autocomplete data in https://github.com/mattermost/mattermost-plugin-github/blob/793631adb358f29a7d1a187dfedd13c2d5dfdf11/server/plugin/command.go#L518. Could you please update that?

**Comment by Remakh** [comment]
> @Remakh I was referring to the autocomplete data in
> 
> https://github.com/mattermost/mattermost-plugin-github/blob/793631adb358f29a7d1a187dfedd13c2d5dfdf11/server/plugin/command.go#L518
> 
> . Could you please update that?

Oh right. 
It's been updated now. Please let me know if there are any more issues.

**Comment by hanzei** [comment]
I think the point from @mickmister in https://github.com/mattermost/mattermost-plugin-github/pull/392#discussion_r508931423 was to disallow the _creation_ of a subscription with both flags set.

**Comment by Remakh** [comment]
> I think the point from @mickmister in [#392 (comment)](https://github.com/mattermost/mattermost-plugin-github/pull/392#discussion_r508931423) was to disallow the _creation_ of a subscription with both flags set.

I've changed a couple things. Please let me know what you think.

File: server/plugin/command.go [hunk 3]
```diff
@@ -625,3 +631,12 @@ func parseCommand(input string) (command, action string, parameters []string) {
 
 	return command, action, parameters
 }
+
+func Contains(a []string, x string) bool {
```
Comment (mickmister) [inline server/plugin/command.go:635]: Maybe to differentiate from `strings.Contains`

```suggestion
func SliceContainsString(a []string, x string) bool {
```

**Review by mickmister — APPROVED** [review]
LGTM :+1: Thanks @Remakh!

I have one suggestion on function naming

File: server/plugin/command.go (hunk 3 above)
Comment (Remakh) [reply server/plugin/command.go]: Thanks for the suggestion! Changed it now

File: server/plugin/command.go [hunk 4]
```diff
@@ -289,6 +291,10 @@ func (p *Plugin) handleSubscribesAdd(_ *plugin.Context, args *model.CommandArgs,
 		} else if len(optionList) == 1 {
 			features = optionList[0]
 			fs := strings.Split(features, ",")
+			if SliceContainsString(fs, "issues") && SliceContainsString(fs, "issue_creations") {
```
Comment (hanzei) [inline server/plugin/command.go:294]: Nit pick: Let's use the `const`s
```suggestion
			if SliceContainsString(fs, featureIssues) && SliceContainsString(fs, featureIssueCreation) {
```

File: server/plugin/command.go [hunk 5]
```diff
@@ -289,6 +291,10 @@ func (p *Plugin) handleSubscribesAdd(_ *plugin.Context, args *model.CommandArgs,
...
+				msg := "Feature list cannot contain both issue and issue_creations"
+				return msg
```
Comment (hanzei) [inline server/plugin/command.go:296]: Nit pick: Is the variable needed?

**Review by hanzei — APPROVED** [review]
Two nit picks, but the rest looks good

**Comment by hanzei** [comment]
@Remakh I think you added an unrelated file `y` in your last commit.

**Comment by Remakh** [comment]
Y has been removed.

**Review by hanzei — APPROVED** [review]

**Comment by DHaussermann** [comment]
@Remakh Thanks for the PR. I have done some testing and `issue_creations` feature you have added seems to work as expected. I have a couple points I'd like to follow up on here.
1. It seems that now `issue_creations` will be added as part of the default when you create a subscriptions without specifying any features. Is this intentional? It seems redundant as `issues` is already part of the default set so you now end up with `pulls,issues,creates,deletes,issue_creations`
2. `issue_creations` is not shows in the list of available features under the help. It should be listed there below `issues`

**Comment by Remakh** [comment]
Thanks for the review! Regarding the default, I believe I misread the code and it wasn't intentional. I can fix that no problem. As for help, I believe I missed that, and I'll add that as well.

**Comment by jfrerich** [comment]
Thanks for the PR, @Remakh.  Looks like your PR has passed all developer reviews and is just waiting on QA.  FYI, they are pretty busy right now with hackathon and release testing, but will get to you soon!

**Comment by Remakh** [comment]
Thank you for the update @jfrerich !
I was really happy I could contribute to this project. If there are any more opportunities in the future I will be sure to look out for them.

**Review by mickmister — APPROVED** [review]
New changes LGTM :+1:

**Comment by jasonblais** [comment]
@DHaussermann this is now ready for QA review

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- New feature works as expect
- Regression testing around subscription features found no issues
- Both points above have now been addressed |

LGTM!

Thanks @Remakh this enhancement really improves the usability. Much appreciated! 

Release tests added to cover this feature.

**Pull request merged**